aspnet/mvc/views/forms/sample/final/Controllers/DemoRegisterController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using FormsTH.ViewModels;
+using Microsoft.AspNet.Mvc;
+
+namespace FormsTH.Controllers
+{
+    // This controller is used only to demonstrate working with forms.
+    public class DemoRegController : Controller
+    {
+        static string _message;
+        public IActionResult Register()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Register(RegisterViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                _message = "HttpPost Register success";
```

Comment:
Removed all the `HttpPost` from this controller except the first. I've had many customers request the `HttpPost` so they can see the complete process, model, etc.

replaced static with `routeValues` object - which is needed in the other controller (to verify selections are working as expected)